MapleServer2/PacketHandlers/Game/BeautyHandler.cs

```diff
@@ -138,7 +138,20 @@ private static void HandleNewBeauty(GameSession session, PacketReader packet)
             Item beautyItem = new Item(beautyItemId) { Color = equipColor, IsTemplate = false };
             BeautyMetadata beautyShop = BeautyMetadataStorage.GetCosmeticShopByItemId(beautyItem.Id);
 
-            HandlePay(session, beautyShop, useVoucher);
+            if (useVoucher)
+            {
+                if(!PayWithVoucher(session, beautyShop))
+                {
+                    return;
+                }
+            }
+            else
+            {
+                if(!PayWithShopItemTokenCost(session, beautyItemId))
+                {
+                    return;
+                }
+            }
```

Comment:
Why not use `HandlePay` here? 🤔

Reply:
This is the only instance where it's using `PayWithShopItemTokenCost`. I had `HandlePay` renamed to `HandleShopPay`